Source/Foundation/bsfUtility/Prerequisites/BsRTTIPrerequisites.h

```diff
@@ -262,6 +262,76 @@ namespace bs
 		}
 	};
 
+	/**
+	 * RTTIPlainType for std::list.
+	 *
+	 * @see		RTTIPlainType
+	 */
+	template<class T> struct RTTIPlainType<std::list<T, StdAlloc<T>>>
+	{
+		enum { id = TID_List }; enum { hasDynamicSize = 1 };
+
+		/** @copydoc RTTIPlainType::toMemory */
+		static void toMemory(const std::list<T, StdAlloc<T>>& data, char* memory)
+		{
+			UINT32 size = sizeof(UINT32);
+			char* memoryStart = memory;
+			memory += sizeof(UINT32);
+
+			UINT32 numElements = (UINT32)data.size();
+			memcpy(memory, &numElements, sizeof(UINT32));
+			memory += sizeof(UINT32);
+			size += sizeof(UINT32);
+
+			for(const auto& item : data)
+			{
+				UINT32 elementSize = rttiGetElemSize(item);
+				RTTIPlainType<T>::toMemory(item, memory);
+
+				memory += elementSize;
+				size += elementSize;
+			}
+
+			memcpy(memoryStart, &size, sizeof(UINT32));
```

Comment:
While its following the pattern of the other containers, do these size prefixes add any value?

Reply:
I was wondering the same as it's not really used in the `fromMemory`-functions. Maybe it's just that *all* saved types have it, so it's better to stay consistent, but @BearishSun should be able to say more about that.

Reply:
Sizes must be written for all dynamically sized plain types, it won't work otherwise.